src/tests/lossy_link.cpp

```diff
@@ -0,0 +1,13 @@
+#include "lossy_link.h"
+
+LossyLink::LossyLink(int seed_num) : _seed({seed_num}) {}
+
+bool LossyLink::drop(float ratio)
+{
+    std::lock_guard<std::mutex> lock(_mutex);
```

Comment:
You don't need the lock anymore.

Reply:
Well, not for our current usage but potentially still.

Reply:
I wouldn't add precautionary locking, you could do that almost everywhere. We should do it only where needed, as it adds unnecessary overhead, both in code and runtime.

Reply:
Ooookay.

Reply:
Done in b0ac231ccd0db3a8e7fadad4784c6a4f21c3ef1a.